release.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -e
+
+[ $# == 1 ] || { echo "usage: $0 version" && exit 1; }
+
+VER=$1
+
+[[ "$VER" =~ ^v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$ ]] || {
+	echo "malformed version: \"$VER\""
+	exit 2
+}
+
+if test -n "$(git ls-files --others | \
+	grep --invert-match '\(vendor\|build/operator-sdk-v.\+\)')";
+then
+	echo "directory has untracked files"
+	exit 1
+fi
+
+if ! $(git diff-index --quiet HEAD --); then
+	echo "directory has uncommitted files"
+	exit 1
+fi
+
+git tag --sign --message "Operator SDK $VER" "$VER"
```

Comment:
There is one important step that's needs to be done before `release.sh` can be used to cut the tag and build the binary.
See the PR we need to merge before cutting a release https://github.com/operator-framework/operator-sdk/pull/457/files

The `Gopkg.toml` template file should be updated to use that version(tag) of the SDK.
Another thing to update is the `version/version.go`. 
I know we do that above in the makefile on `make release`.
But we need to do that before we cut the tag.

Otherwise version.go could be:
```Go
package version

var (
	Version = "0.0.5+git"
)
```
And if we do `release.sh v0.0.6` the tag will be cut before the version.go file is updated and any operator vendoring the `v0.0.6` tag will see the older version on doing the following:
```Go
import sdkVersion "github.com/operator-framework/operator-sdk/version"

log.Printf("operator-sdk Version: %v", sdkVersion.Version)
```

So release.sh needs to do a check to make sure that the `Gopkg.toml` template and `version/version.go` file are correctly updated to the desired version before cutting the tag and building the binary.

Reply:
I'll write a script to make the substitutions necessary before tagging and building.